added: Buffer.loadChannelDialog and its corresponding help documentation

prko opened the pull request:
## Purpose and Motivation
It would be convenient for SuperCollider to provide a counterpart for `Buffer.readChannel`, as it already provides `Buffer.loadDialog` as the counterpart to `Buffer.read`.
   - A new class method for Buffer is added: `Buffer.loadChannelDialog`.
   - The appropriate part for the new method above to `Buffer.schelp` file is added.
   - Some text lines are arranged to avoid wrong automatic indentation.

## Types of changes

- Documentation
- New feature

## To-do list

<!-- Complete an item by checking it: [x]. Add new entries to track your progress -->

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

cdbzb reviewed (commented):
Hi @prko thanks for this! I support the idea.

I do have a comment regarding the examples in the documentation. 

My understanding is that code blocks should *not* be indented: ie

```
(
a = 3; \\right
)
```
not
```
(
    a = 3; //wrong!
)
```

so these should be reverted!

prko commented:
@cdbzb I did it! Thanks!

cdbzb commented:
Wow that was fast! thank you!

while we're here: I think this (and similar!)

```
		^super.newCopyArgs(server,
			bufnum,
			numFrames,
			numChannels).sampleRate_(server.sampleRate).cache
```
would be better
```
		^super.newCopyArgs(
			server,
			bufnum,
			numFrames,
			numChannels
		).sampleRate_(server.sampleRate).cache
```

and from my understanding of the code style guidelines https://github.com/supercollider/supercollider/wiki/Code-style-guidelines#method-calls

```
		^super.newCopyArgs(server, bufnum)
		.doOnInfo_(action).cache
		.allocReadChannel(path, startFrame, numFrames, channels,
```

should be indented like

```
		^super.newCopyArgs(server, bufnum)
			.doOnInfo_(action).cache
			.allocReadChannel(path, startFrame, numFrames, channels,
```

cdbzb commented:
I'd also suggest that you save the changes in the Line Layout Help for a different PR - its better to segregate PRs by purpose.

...or better you might separate it into one *functional* PR (adding a method) which requires testing etc. and another PR for various reformatting and cleanup efforts. It makes it easier for us to test and review, and also makes the history easier to understand if that makes sense.

cdbzb commented:
looking forward to testing the actual method! but that's it for my day today - more tomorrow!

prko commented:
@cdbzb  I changed the parts in the part `Buffer.sc` according to your suggestion, and reverted the file `LineLayout.schelp`. I will make a new PR "increased readability and comfortable code evaluation" later with other files. I am not sure when I can do it.

telephon commented on HelpSource/Classes/Buffer.schelp:
```diff
@@ -358,7 +358,32 @@ b = Buffer.loadDialog(s, action: { arg buffer;
-
+		
+method:: loadChannelDialog
+As link::#*readChannel:: above, but gives you a load dialog window to browse for a file. Cocoa compatible.
```
you could drop "Cocoa compatible."? – it should be compatible to all platforms that sclang supports.

prko replied:
done! Also removed the corresponding part in `.loadDialog`

telephon reviewed (approved):
Since this matches the second file reading operation and simple enough, I think it is justified.

cdbzb reviewed (commented):
Tetsted on Mac - works correctly

prko commented on SCClassLibrary/Common/Control/Buffer.sc:
```diff
@@ -186,11 +189,11 @@ Buffer {
-								{("Could not delete data file:" + path).warn;});
-							action.value(buf);
-						}).allocRead(path, 0, -1, {|buf| ["/b_query", buf.bufnum] })
+					.cache.doOnInfo_({ |buf|
```
done!

prko commented on SCClassLibrary/Common/Control/Buffer.sc:
```diff
@@ -88,35 +91,35 @@ Buffer {
-					.doOnInfo_(action).cache
-					.allocReadChannel(path, startFrame, numFrames, channels,
-						{|buf|["/b_query", buf.bufnum]})
+		.doOnInfo_(action).cache
```
done!

prko commented:
I have tested the code block in the help document and it worked on MacOS 13.6.3 (silicon), Windows 11 Pro (arm64) and Ubuntu 22.04 (arm64).

cdbzb commented on SCClassLibrary/Common/Control/Buffer.sc:
```diff
@@ -673,7 +675,20 @@ Buffer {
 		Dialog.openPanel({ arg path;
 			buffer.doOnInfo_(action)
-				.allocRead(path, startFrame, numFrames, { ["/b_query", buffer.bufnum] })
+			.allocRead(path, startFrame, numFrames, { ["/b_query", buffer.bufnum] })
```
sorry @prko I missed one! this one should be indented also - then we're good to go!

prko replied:
Done, I hope I correctly understood you

cdbzb reviewed (approved):
thanks!
